lib/fog/utilities.rb

```diff
@@ -1,7 +1,9 @@
 class Hash
   def symbolize_keys!
     keys.each do |key|
-      self[(key.to_sym rescue key)] = delete(key) if key.respond_to?(:to_sym) && !key.is_a?(Fixnum)
+      if key.respond_to?(:to_sym) && !key.is_a?(Integer)
```

Comment:
`Integer` does not respond to `to_sym` thus you might not need this second check.

On another note i've been thinking about this code since i've extracted this provider. It feels like a hack to me where we might not neet at all if we work with string keys instead. What do you think?